### PR TITLE
Re-thinking available features

### PR DESCRIPTION
**Pull request by arichiv**
Realized this when working on the design doc, let me know what you think.

### CODE REVIEW

**Comment by johannhof** [comment]
I think I'm on board with the opener part, though I'm wondering if cross-browser folks have any thoughts (@bvandersloot-mozilla or @annevk for example).

I don't fully grok `popin must know the referrer to gain access to the correct storage partition`. You mean internally the browser needs to know? We could still decide not to expose the full referrer if the site doesn't want us to, right?

**Comment by arichiv** [comment]
We could decide not to expose the referrer, but at least the origin is sort of known given the Popin-Policy header.

**Comment by johannhof** [comment]
That's a good point, so we at least leak the origin of the referrer. I'm still not convinced that this would mean we have to forbid usage of noreferrer. How does this interact with [Referrer-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy)? Do we enforce the referrer even though the policy says no-referrer? While that would most accurately reflect the privacy properties of the pop-in it seems a bit problematic from an API standpoint.

And if the developer choose a policy of unsafe-url but chooses to select no-referrer on specific window.open calls, sending the full referrer URL on those calls might actually regress on user privacy.

OTOH it seems like [rel=noreferrer implies rel=noopener](https://html.spec.whatwg.org/multipage/links.html#link-type-noreferrer) 🤔, which might not be what we want.

Soo maybe we enforce a Referer header that is always the origin of the referrer?

File: README.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ Investigations into [protecting openers](https://groups.google.com/a/chromium.or
...
 
-* popin - Indicates the window should be opened as an inline popin. The target must be `_blank` as others have no special meaning in this context. Cannot be used with `popup` feature as these options conflict. The features `width`, `height`, `left`, and `top` may or may not be respected by the rendering browser (especially on mobile). `noopener` and `noreferrer` will be respected as usual.
+* popin - Indicates the window should be opened as an inline popin. The target must be `_blank` as others have no special meaning in this context. Cannot be used with `popup` feature as these options conflict. The features `width`, `height`, `left`, and `top` may or may not be respected by the rendering browser (especially on mobile). It will not be possible to use `noopener` or `noreferrer` (attempting to do so will cause rejection of popin creation) as the popin must know the referrer to gain access to the correct storage partition and the blocking nature of popins negates any utility in having unreachable popins (a window would want to know when they were closed).
```
Comment (krgovind) [inline README.md:83]: > and the blocking nature of popins negates any utility in having unreachable popins

This bit is hard to parse. Could you help me understand what it's trying to say?

Comment (arichiv) [reply README.md:83]: (1) noopener is a bad idea because of the properties of Popins. Specifically, they block access to the opening window until closed and the opened window has some knowledge of who opened them as they can require only specific origins do so
(2) noreferrer might be a lost cause? It's true that info is restricted, but at least some knowledge about the origin of the referrer is known

@johannhof's point is that we can't just casually violate the Referrer-Policy and in some cases noreferrer seems to imply nooepner

Perhaps we just have to allow the interaction to be weird, and allow them as-is
